SOS/APP/app.c

```diff
@@ -0,0 +1,231 @@
+/*
+* app.c
+*
+* Created: 5/30/2023 2:25:22 PM
+*  Author: Alaa
+*/
+
+#include "interrupts.h"
+
+#include "led_interface.h"
+#include "btn_interface.h"
+#include "sos_interface.h"
+
+#include "app.h"
+
+static void app_led0(void);
+static void app_led1(void);
+static void app_start_btn(void);
+static void app_stop_btn(void);
+
+static str_btn_config_t_ gl_str_start_btn;
+static str_btn_config_t_ gl_str_stop_btn;
+
+static str_sos_task_t_ gl_str_led0_task;
+static str_sos_task_t_ gl_str_led1_task;
+static str_sos_task_t_ gl_str_stop_btn_task;
+
+static enu_btn_state_t_ gl_enu_start_btn_state = BTN_STATE_NOT_PRESSED;
+static enu_btn_state_t_ gl_enu_stop_btn_state = BTN_STATE_NOT_PRESSED;
+
+
+enu_app_init_status_t_ app_init(void)
+{
+	
+	/* Initialize LEDs */
+	enu_led_error_t_ lo_enu_led_status = 0;
+	lo_enu_led_status = led_init(APP_LED_0_PORT, APP_LED_0_PIN);
+	lo_enu_led_status += led_init(APP_LED_1_PORT, APP_LED_1_PIN);
+	
+	if(lo_enu_led_status != LED_OK) {return APP_INIT_FAILED;}
+	
+	/* Initialize push buttons */
+	gl_str_start_btn = (str_btn_config_t_)
+	{
+		.enu_btn_port	    = APP_START_ENU_BTN_PORT		,
+		.enu_btn_pin	    = APP_START_ENU_BTN_PIN 		,
+		.enu_btn_read_mode  = BTN_MODE_INTERRUPT			,
+		.enu_btn_pull_type  = BTN_EXTERNAL_PULL_DOWN		,
+		.enu_btn_activation = BTN_ACTIVATED
+	};
+	
+	gl_str_stop_btn = (str_btn_config_t_)
+	{
+		.enu_btn_port	    = APP_STOP_ENU_BTN_PORT			,
+		.enu_btn_pin	    = APP_STOP_ENU_BTN_PIN 			,
+		.enu_btn_read_mode  = BTN_MODE_INTERRUPT			,
+		.enu_btn_pull_type  = BTN_EXTERNAL_PULL_DOWN		,
+		.enu_btn_activation = BTN_ACTIVATED
+	};
+	
+	enu_btn_status_code_t_ lo_enu_btn_status;
+	
+	lo_enu_btn_status = btn_init(&gl_str_start_btn);
+	lo_enu_btn_status += btn_set_notification(&gl_str_start_btn, app_start_btn);
+	lo_enu_btn_status += btn_init(&gl_str_stop_btn);
+	
+	if(lo_enu_btn_status != BTN_STATUS_OK) {return APP_INIT_FAILED;}
+	
+	/* Initialize the SOS */
+	enu_sos_status_t_ lo_enu_sos_status = sos_init();
+	
+	if(lo_enu_sos_status != SOS_STATUS_SUCCESS) {return APP_INIT_FAILED;}
+	
+	/* Create Tasks */
+	 
+	gl_str_led0_task = (str_sos_task_t_)
+	{
+		.uint8_task_id			 = NULL		 ,
+		.uint8_task_priority	 = 1		 ,
+		.uint16_task_periodicity = 15		 ,
+		.ptr_func_task			 = app_led0
+	};
+	
+	gl_str_led1_task = (str_sos_task_t_)
+	{
+		.uint8_task_id			 = NULL		 ,

```

Comment:
Invalid Initialization Note: 
Task ID Shall be assigned to a ZERO_INIT Value instead of NULL. As this is not a pointer

Reply:
@MinaRaoufAziz 

`std.h@32,33`

```
#define NULL        (0)
#define NULL_PTR    ((void *) 0)
```

`NULL` in our std header file `LIB/std.h` refers to `0`, are you referring to that or just that we update it to a more appropriate naming?

---
@AalaaIbrahim